pkg/gitreceive/build.go

```diff
@@ -0,0 +1,295 @@
+	if err := ioutil.WriteFile(finalManifestFileLocation, []byte(finalManifest), os.ModePerm); err != nil {
+		return fmt.Errorf("writing final manifest %s (%s)", finalManifestFileLocation, err)
+	}
+
+	configDir := "/var/minio-conf"
+	if err := os.MkdirAll(configDir, os.ModePerm); err != nil {
+		return fmt.Errorf("creating minio config file (%s)", err)
+	}
+
+	configCmd := mcCmd(configDir, "config", "host", "add", fmt.Sprintf("%s://%s:%s", storage.schema(), storage.host(), storage.port()), creds.key, creds.secret)
+	if err := run(configCmd); err != nil {
+		return fmt.Errorf("configuring the minio client (%s)", err)
+	}
+
+	makeBucketCmd := mcCmd(configDir, "mb", fmt.Sprintf("%s://%s:%s/git", storage.schema(), storage.host(), storage.port()))
+	// Don't look for errors here. Buckets may already exist
+	// https://github.com/deis/builder/issues/80 will eliminate this distaste
+	run(makeBucketCmd)
+
+	cpCmd := mcCmd(configDir, "cp", fmt.Sprintf("%s.tar.gz", appName), tarURL)
+	cpCmd.Dir = repoDir
+	if err := run(cpCmd); err != nil {
+		return fmt.Errorf("copying %s.tar.gz to %s (%s)", appName, tarURL, err)
+	}
+
+	log.Info("Starting build")
+	log.Debug("Starting pod %s", buildPodName)
+	kCreateCmd := exec.Command(
+		"kubectl",
+		fmt.Sprintf("--namespace=%s", conf.PodNamespace),
+		"create",
+		"-f",
+		finalManifestFileLocation,
+	)
+	if log.IsDebugging {
+		kCreateCmd.Stdout = os.Stdout
+	}
+	kCreateCmd.Stderr = os.Stderr
+	if err := run(kCreateCmd); err != nil {
+		return fmt.Errorf("creating builder pod (%s)", err)
+	}
+
+	// poll kubectl every 100ms to determine when the build pod is running
+	// TODO: use the k8s client and watch the event stream instead (https://github.com/deis/builder/issues/65)
+	for {
+		cmd := kGetCmd(conf.PodNamespace, buildPodName)
+		var out bytes.Buffer
+		cmd.Stdout = &out
+		// ignore errors
+		run(cmd)
+		outStr := string(out.Bytes())
+		if strings.Contains(outStr, "phase: Running") {
+			break
+		} else if strings.Contains(outStr, "phase: Failed") {
+			return fmt.Errorf("build pod %s entered phase: Failed", buildPodName)
+		}
+		time.Sleep(100 * time.Millisecond)
+	}
+
+	// get logs from the builder pod
+	kLogsCmd := exec.Command(
+		"kubectl",
+		fmt.Sprintf("--namespace=%s", conf.PodNamespace),
+		"logs",
+		"-f",
+		buildPodName,
+	)
+	kLogsCmd.Stdout = os.Stdout
+	if err := run(kLogsCmd); err != nil {
+		return fmt.Errorf("running %s to get builder logs (%s)", strings.Join(kLogsCmd.Args, " "), err)
+	}
+
+	// poll the s3 server to ensure the slug exists
+	for {
+		// for now, assume the error indicates that the slug wasn't there, nothing else
+		// TODO: implement https://github.com/deis/builder/issues/80, which will clean this up siginficantly
+		lsCmd := mcCmd(configDir, "ls", pushURL)
+		if err := run(lsCmd); err == nil {
+			break
+		}
+	}
+
+	log.Info("Build complete.")
+
+	log.Info("Launching...")
+
+	buildHook := &pkg.BuildHook{
+		Sha:         gitSha,
+		ReceiveUser: conf.Username,
```

Comment:
I'm assuming you mean the line above. Unfortunately I can't - workflow sends back a `400 Bad Request` when I do...